### ISSUE TITLE
Can we add examples of using metrics package in PyTorch Lightning workflow?

### ISSUE THREAD

**Issue by amitness** [PyTorchLightning/pytorch-lightning#2265]
Hi,

I am really excited about the release of the metrics package. Can we add more examples of using it in the context of the lightning module?

For example, if I want to calculate the global accuracy/f1-score at the end of each epoch, is doing it this way correct? I feel it's wrong as averaging batch-f1 to get the global average can be misleading. When I tested it on a toy example and compared it to a global micro/macro/weighted f1-score, the f1-scores were not the same.

This was my initial attempt:
```python
def training_step(self, batch, batch_idx):
    criterion = nn.CrossEntropyLoss()
    x, y = batch
    logits = self(x)
    loss = criterion(logits, y)
    batch_f1 = f1_score(logits.argmax(dim=-1), y)
    return {'loss': loss, 'f1': batch_f1}
```
and then averaging at the end of each epoch
```python
def training_epoch_end(self, outputs):
    avg_loss = torch.stack([x['loss'] for x in outputs]).mean()
    f1 = torch.stack([x['f1'] for x in outputs]).mean() * 100
    logs = {'avg_train_loss': avg_loss, 'f1': f1}
    return {'avg_train_loss': avg_loss,
            'progress_bar': logs, 'log': logs}
```

**Comment by amitness** [comment]
@williamFalcon

**Comment by williamFalcon** [comment]
good suggestion.
For validation the metrics should just happen on epoch_end. Each val_step should pass the necessary tensors to calculate the "global" metric.

But we have a new way of doing this which is simpler and will be release in the next few weeks

**Comment by amitness** [comment]
Awesome. Looking forward to it then.
